cmd/kubeapps-apis/plugins/resources/v1alpha1/server.go

```diff
@@ -103,14 +103,34 @@ func (s *Server) GetResources(r *v1alpha1.GetResourcesRequest, stream v1alpha1.R
 		return err
 	}
 
+	var resourcesToReturn []*pkgsGRPCv1alpha1.ResourceRef
+	// If the request didn't specify a filter of resource refs,
+	// we return all those found for the installed package. Otherwise
+	// we only return the requested ones.
+	if len(r.GetResourceRefs()) == 0 {
+		resourcesToReturn = refsResponse.GetResourceRefs()
+	} else {
```

Comment:
wrt my previous PR comment - Ah, ok, got it, this 400 is thrown if filtering and nothing is found. 
However... why not a 404-like? I mean, a 400 would assume the client is doing sth wrong, but it might haven caused by querying a transient state, no?

Reply:
Yes, currently it's intended to catch a bad request, but I may have to update the behaviour if we do have the transient situations where the request is good but the refs aren't yet available. In that situation, if wait=true, I'd hold off and keep waiting. But I'll update that when testing IRL from the UI. Thanks!